waspc/data/Generator/templates/server/src/routes/apis/index.ts

```diff
@@ -2,18 +2,35 @@
 import express from 'express'
 import prisma from '../../dbClient.js'
 import { handleRejection } from '../../utils.js'
+import { getDefaultMiddleware, toMiddlewareArray } from '../../middleware.js'
 {=# isAuthEnabled =}
 import auth from '../../core/auth.js'
 import { type UserInContext } from '../../_types'
 {=/ isAuthEnabled =}
 
+const idFn = (x: any) => x
+
 {=# apiRoutes =}
 {=& importStatement =}
+
+{=# routeMiddlewareConfigFnDefined =}
+{=& routeMiddlewareConfigFnImportStatement =}
+{=/ routeMiddlewareConfigFnDefined =}
+{=^ routeMiddlewareConfigFnDefined =}
+const {=& routeMiddlewareConfigFnImportAlias =} = idFn
+{=/ routeMiddlewareConfigFnDefined =}
+
 {=/ apiRoutes =}
 
 const router = express.Router()
 
+/*
+  NOTE: Since some middleware needs to run _before_ the method, like CORS for GET/POST requests,
+  we need to use `Router.use`. However, since this applies to all methods for that route, this
+  means we can only support _unique_ route paths.
+*/
```

Comment:
This confuses me a bit, I would think that we can somehow go about supporting different methods.

Problem is actually that middleware is applied per route, not per method, is that it?
Maybe then we also shouldn't apply it per method in Wasp, but apply it per route instead? You know, have something like `server-route "/foo/bar" { middlewareFn: ... }`? I made up this syntax, but I guess something like this? This does complicate things a bit though hm, but maybe is a direction we should actually explore?

Reply:
This is no longer needed with `namespace`, which is similar to your `server-route`. Open to tweaking that name/properties.